bluemira/magnetostatics/baseclass.py

```diff
@@ -192,9 +192,84 @@ def plot(self, ax: Optional[Axes] = None, show_coord_sys: bool = False):
         # Plot local coordinate system
         if show_coord_sys:
             ax.scatter(*self.origin, color="k")
-            ax.quiver(*self.origin, *self.dcm[0], length=self.breadth, color="r")
-            ax.quiver(*self.origin, *self.dcm[1], length=self.length, color="r")
-            ax.quiver(*self.origin, *self.dcm[2], length=self.depth, color="r")
+            ax.quiver(*self.origin, *self.dcm[0], length=1, color="r")
+            ax.quiver(*self.origin, *self.dcm[1], length=1, color="r")
+            ax.quiver(*self.origin, *self.dcm[2], length=1, color="r")
+
+
+class RectangularCrossSectionCurrentSource(CrossSectionCurrentSource):
+    """
+    Abstract base class for a current source with a rectangular cross-section.
+    """
+
+    breadth: float
+    depth: float
+    length: float
+
+
+class PolyhedralCrossSectionCurrentSource(CrossSectionCurrentSource):
+    """
+    Abstract base class for a current source with a polyhedral cross-section.
+    """
+
+    face_points: np.ndarray
+    face_normals: np.ndarray
+
+    def rotate(self, angle: float, axis: Union[np.ndarray, str]):
+        """
+        Rotate the CurrentSource about an axis.
+
+        Parameters
+        ----------
+        angle:
+            The rotation degree [degree]
+        axis:
+            The axis of rotation
+        """
+        super().rotate(angle, axis)
+        r = rotation_matrix(np.deg2rad(angle), axis).T
+        self.face_normals = np.array([n @ r for n in self.face_normals])
+        self.face_points = np.array([p @ r for p in self.face_points])
+        self.mid_points = np.array([p @ r for p in self.mid_points])
```

Comment:
I think this whole functionality needs a profile and then I'd like to do all the speed-ups in a single PR. Thing is, it's quite possible it would be better to move to a different datastructure to do the maths faster, and that might be a good candidate for some hardcore einsumming. The rotations are not concerning at the moment. If they are, there might actually be cute tricks to rotate the field points rather than the sources in the `HelmholtzCage` but that's another story.